fabric-models-v0/src/main/java/net/fabricmc/fabric/api/client/model/ModelAppender.java

```diff
@@ -20,8 +20,26 @@
 
 import net.minecraft.client.util.ModelIdentifier;
 import net.minecraft.resource.ResourceManager;
+import net.minecraft.util.Identifier;
 
 @FunctionalInterface
 public interface ModelAppender {
+	/**
+	 * Provides an opportunity inform the game that you would like it to load and bake a model,
+	 * even if that model is not used by any blocks or items.
+	 * @param out Accepts paths to be loaded. Arguments that are {@link ModelIdentifier} will be
+	 *            loaded through the blockstate JSON system or, if the variant is {@code inventory}, the item model folder.
+	 *            Otherwise, the argument is directly loaded as a JSON.
+	 *            For example, <pre>new Identifier("mymod", "foo/bar")</pre> will request loading of the file
+	 *            <pre>/assets/mymod/models/foo/bar.json</pre>
+	 */
+	default void appendAllIdentifiers(ResourceManager manager, Consumer<Identifier> out) {
```

Comment:
this bothers me in that lambdas still default to the older/deprecated method. Should I just make a new interface for this and have the old interface extend it?

Reply:
I believe that may be the best option. I am not going to call myself a master at using json models but within the api structure that would make the most sense since it would allow any old implementations which just move over to the new interface and change the generic to continue working.